src/store/cartState.ts

```diff
@@ -0,0 +1,39 @@
+import { create } from 'zustand'
+import { Product } from '../model/product'
+
+interface CartItem extends Product {
```

Comment:
Just a suggestion, you can rather store just the product Id instead of the whole product and then deprive the actual product if needed. The issue with storing the whole product in state is what if the price or something is updated in the backend, the user won't be able to see that change in their cart

Reply:
good idea. will do that in a refactor, just to keep the features rolling 👍